scrape/scrape.go

```diff
@@ -1641,22 +1658,23 @@ loop:
 				sl.metrics.targetScrapePoolExceededLabelLimits.Inc()
 				break loop
 			}
-
-			// Append metadata for new series if they were present.
-			updateMetadata(lset, true)
 		}
 
 		if seriesAlreadyScraped {
 			err = storage.ErrDuplicateSampleForTimestamp
 		} else {
-			if ctMs := p.CreatedTimestamp(); sl.enableCTZeroIngestion && ctMs != nil {
+			ctMs := p.CreatedTimestamp()
+			if sl.enableCTZeroIngestion && ctMs != nil {
```

Comment:
`enableCTZeroIngestion` check should be before we call `p.CreatedTimestamp()`

Reply:
Not sure if that's what we want here to be honest. The feature flag is meant to be used to ingest zeros. I was hoping that adding to metadata is safe enough to be added without feature flag.

I understand that OM-text parser performance is not great for CreatedTimestamp, maybe working on performance is a pre-requisite to the work here 🤔

Reply:
OM Text parser performance is one thing which deserves a flag (we can't really optimize it without OM 2.0), but I think we could try to design how flags will look like with other CT related features:

* zero samples (I think this should be a separate flag)
* CT in storage (WAL) including CT in remote write
* CT used in PromQL e.g. optimized rate and resets (perhaps no need for flag, but slow iterations?)
* CT in TSDB storage